### PR TITLE
Adding Cilium to create_stack

### PR DESCRIPTION
**Pull request by ahsan518**
This change adds Cilium to create_stack script, and uses crio
by default unless the user otherwise specify

#40 

Signed-Off: Syed

### CODE REVIEW

File: clr-k8s-examples/create_stack.sh [hunk 1]
```diff
@@ -97,7 +103,27 @@ function cni() {
 	curl -o "${CANAL_DIR}/overlays/${CANAL_VER}/canal/rbac.yaml" "$CANAL_URL/rbac.yaml"
 	# canal doesnt pass kustomize validation
 	kubectl apply -k "${CANAL_DIR}/overlays/${CANAL_VER}" --validate=false
-
+	;;
+		cilium)
```
Comment (krsna1729) [inline clr-k8s-examples/create_stack.sh]: Make sure shfmt and shellcheck are happy

https://github.com/mvdan/sh
https://github.com/koalaman/shellcheck

File: clr-k8s-examples/setup_system.sh [hunk 2]
```diff
@@ -10,7 +10,8 @@ HIGH_POD_COUNT=${HIGH_POD_COUNT:-""}
 # set no proxy
 ADD_NO_PROXY="10.244.0.0/16,10.96.0.0/12"
 ADD_NO_PROXY+=",$(hostname -I | sed 's/[[:space:]]/,/g')"
-: "${RUNNER:=containerd}"
+: "${RUNNER:=crio}"
```
Comment (krsna1729) [inline clr-k8s-examples/setup_system.sh]: Keep this separate

Comment (ahsan518) [reply clr-k8s-examples/setup_system.sh]: Done

File: clr-k8s-examples/create_stack.sh (hunk 1 above)
Comment (ahsan518) [reply clr-k8s-examples/create_stack.sh]: Done

File: clr-k8s-examples/setup_system.sh [hunk 3]
```diff
@@ -12,6 +12,7 @@ ADD_NO_PROXY="10.244.0.0/16,10.96.0.0/12"
 ADD_NO_PROXY+=",$(hostname -I | sed 's/[[:space:]]/,/g')"
 : "${RUNNER:=containerd}"
 
+
```
Comment (krsna1729) [inline clr-k8s-examples/setup_system.sh]: remove this file

**Comment by krsna1729** [comment]
squash the commits
use intel email

File: clr-k8s-examples/create_stack.sh (hunk 1 above)
Comment (krsna1729) [reply clr-k8s-examples/create_stack.sh]: I still see changes when i run `shfmt -w -l create_stack.sh`

**Comment by NitinAtIntel** [comment]
Are these done? use intel email, correct signoff?

File: clr-k8s-examples/create_stack.sh (hunk 1 above)
Comment (NitinAtIntel) [reply clr-k8s-examples/create_stack.sh]: Did all the shfmt output get fixed?

Comment (ahsan518) [reply clr-k8s-examples/create_stack.sh]: Done

File: clr-k8s-examples/create_stack.sh [hunk 4]
```diff
@@ -10,6 +10,8 @@ SCRIPT_DIR="$(dirname "${BASH_SOURCE[0]}")"
 : ${MASTER_IP:=}
 HIGH_POD_COUNT=${HIGH_POD_COUNT:-""}
 
+CNI=${CNI:-"canal"}
+RUNNER=${RUNNER:-"crio"}
```
Comment (ganeshmaharaj) [inline clr-k8s-examples/create_stack.sh]: I thought this change was to moved to a different PR. Is that not true? Also the runner settings should happen in `setup_system.sh` .

Comment (krsna1729) [reply clr-k8s-examples/create_stack.sh]: good catch. i thought so too

Comment (krsna1729) [reply clr-k8s-examples/create_stack.sh]: ok so that was in system_setup.sh https://github.com/clearlinux/cloud-native-setup/pull/181

Comment (krsna1729) [reply clr-k8s-examples/create_stack.sh]: Guess we do need RUNNER here because of cilium. but maybe it should reflect current containerd setting in the repo and #181 should update it to crio even for create_system.sh

Comment (ahsan518) [reply clr-k8s-examples/create_stack.sh]: So Runner is set to crio, because we are using Crio as a default based on 181 change, and the i used these here so that the user can specify to use CNI to either be Canal, Cilium, Flannel.

**Comment by ganeshmaharaj** [comment]
This change doesn't work out of the box for our setup.

```
[vagrant@f-01 clr-k8s-examples]$ ./create_stack.sh cni
~/cloud-native-setup/clr-k8s-examples/0-cilium/overlays/v1.6/cilium ~/cloud-native-setup/clr-k8s-examples
--2019-09-25 22:05:33--  https://github.com/cilium/cilium/archive/v1.6.tar.gz
Resolving github.com (github.com)... 192.30.255.112
Connecting to github.com (github.com)|192.30.255.112|:443... connected.
HTTP request sent, awaiting response... 302 Found
Location: https://codeload.github.com/cilium/cilium/tar.gz/v1.6 [following]
--2019-09-25 22:05:34--  https://codeload.github.com/cilium/cilium/tar.gz/v1.6
Resolving codeload.github.com (codeload.github.com)... 192.30.255.120
Connecting to codeload.github.com (codeload.github.com)|192.30.255.120|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: unspecified [application/x-gzip]
Saving to: ‘v1.6.tar.gz’

v1.6.tar.gz                               [              <=>                                                      ]  22.14M  8.25MB/s    in 2.7s

2019-09-25 22:05:37 (8.25 MB/s) - ‘v1.6.tar.gz’ saved [23217369]

~/cloud-native-setup/clr-k8s-examples/0-cilium/overlays/v1.6/cilium/cilium-1.6/install/kubernetes ~/cloud-native-setup/clr-k8s-examples/0-cilium/overlays/v1.6/cilium ~/cloud-native-setup/clr-k8s-examples
./create_stack.sh: line 117: helm: command not found
```
WHy helm??

**Comment by krsna1729** [comment]
> ./create_stack.sh: line 117: helm: command not found
Wait, helm was bundled in cloud-native right?

> WHy helm??
Cilium switched to helm as the way to customize from 1.6

File: clr-k8s-examples/create_stack.sh (hunk 4 above)
Comment (krsna1729) [reply clr-k8s-examples/create_stack.sh]: Ok depending on the ordering of which one to merge first you can figure this out

**Comment by CraigSterrett** [comment]
@ahsan518  Please discuss with Justin the options to use Kustomize here

**Comment by ahsan518** [comment]
Created a new pull request. closing this.
